### PR TITLE
write the "stack" frames during uncaught exceptions

### PR DESCRIPTION
**Pull request by disruptek**

### CODE REVIEW

**Comment by disruptek** [comment]
No opinions?

File: cps/spec.nim [hunk 1]
```diff
@@ -357,6 +349,97 @@ macro cpsMagic*(n: untyped): untyped =
...
+
+proc trampoline*[T: Continuation](c: T): T =
+  ## This is the basic trampoline: it will run the continuation
+  ## until the continuation is no longer in the `Running` state.
+  var c: Continuation = c
+  while not c.isNil and not c.fn.isNil:
+    try:
+      c = c.fn(c)
+    except Exception:
+      writeStackFramesImpl c
```
Comment (alaviss) [inline cps/spec.nim:421]: Why don't we use `writeStackFrames` instead?

**Review by alaviss — COMMENTED** [review]
LGTM for the most part.

File: cps/spec.nim (hunk 1 above)
Comment (disruptek) [reply cps/spec.nim:421]: It's voodoo or whatever so we can't use it directly.

**Pull request merged**